### ISSUE TITLE
Modal: Focus Trap on action buttons

### ISSUE THREAD

**Issue by jeryj** [Automattic/wp-calypso#44665]
There is a focus trap on the action buttons in the support Modal (possibly others as well).

This seems to be due to the action buttons being the first root-level element of the modal content. This wrapping `<div>` is what appears to break it.

https://github.com/Automattic/wp-calypso/blob/4537033c6794449ce0a10b69d72daf5fa658d553/packages/components/src/dialog/index.jsx#L127-L129

Removing that, we would need to add some kind of `overflow-y: auto;` to the `.ReactModal__Content` and position the action bar buttons fixed to the bottom of the content.

I believe the root issue here is that the modal is rightly supposed to trap focus within its contents. The Modal is ignoring the content within the `<div>` as focusable even if I remove the `tabindex=-1` attribute, and only considering the action buttons as its focusable content.

#### Steps to reproduce

1. Starting at URL: `/home`
2. Click a support article help link to open the dialog/modal overlay
3. Tab through the article
4. Once focus is on the action buttons (Close and View Article), focus cannot leave until one of those is selected

#### What I expected
Pressing Tab from the Close and View Article, focus should return to the top of the modal.

#### What happened instead
Focus is trapped within the action buttons.

#### Screenshot / Video
![calypso-modal-focus-trap](https://user-images.githubusercontent.com/967608/89339499-df9df980-d663-11ea-87aa-daba864ac887.gif)
